File: Easy/27. Remove Element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        ArrayList<Integer> newNums = new ArrayList<Integer>(); 
        for(int i = 0; i < nums.length; i++){
            if(nums[i] != val){
                newNums.add(nums[i]);
            }
        }
        
        for(int i = 0; i < newNums.size(); i++){
            nums[i] = newNums.get(i);
        }
        return newNums.size();
    }
}

File: Easy/27. Remove Element/solution.c
int removeElement(int* nums, int numsSize, int val){
    int* newNums = calloc(numsSize, sizeof(int));
    
    int newArrCount = 0;
    for(int i = 0; i < numsSize; i++){
        int n = nums[i];
        if(n != val){
            newNums[newArrCount] = n;
            newArrCount++;
        }
    }
    
    for(int i = 0; i < numsSize; i++){
        nums[i] = newNums[i];
    }
    
    return newArrCount;
}